### PR TITLE
Adds isLocal and isRemote

### PR DESCRIPTION
**Pull request by micheleorselli**
As discussed in #2 here is a first implementation. Here are a couple of things I'd like to discuss 
- is there a local range for ipv6 addresses? the function now checks only for ipv4
- are the ipv4 private range complete? I'm not totally sure 
  @akalongman let me know what do you think about that

### CODE REVIEW

File: src/Ip.php [hunk 1]
```diff
@@ -74,6 +74,37 @@ public static function isValidv6($ip)
...
+     * @return boolean     true if the IP is local, otherwise false
+     */
+    public static function isLocal($ip)
+    {
+        if (self::isValidv4($ip)) {
+            return self::match($ip, '10.*.*.*') ||
+                   self::match($ip, '127.*.*.*') ||
+                   self::match($ip, '192.168.*.*') ||
+                   self::match($ip, '169.254.*.*') ||
+                   self::match($ip, '172.16.*.*') ||
```
Comment (moebrowne) [inline src/Ip.php]: `172.16.*.*` is a little specific it should be `172.16/12` aka `172.16.0.0 - 172.31.255.255` ([reference](https://tools.ietf.org/html/rfc1918#page-4))

File: src/Ip.php [hunk 2]
```diff
@@ -74,6 +74,37 @@ public static function isValidv6($ip)
...
     /**
+     * Checks if an IP is local
+     *
+     * @param  string  $ip IP
+     * @return boolean     true if the IP is local, otherwise false
+     */
+    public static function isLocal($ip)
+    {
+        if (self::isValidv4($ip)) {
+            return self::match($ip, '10.*.*.*') ||
```
Comment (moebrowne) [inline src/Ip.php]: Rather than calling `self::match()` repeatedly it might be cleaner to define an array of local IP patterns and loop over them

**Review by moebrowne — CHANGES_REQUESTED** [review]

**Review by moebrowne — COMMENTED** [review]
In terms of IPv6 local addresses ranges:
- `fe80::/10`
- `::1/128`
- `fc00::/7`

**Comment by micheleorselli** [comment]
thank you for the feedback, updating the PR soon 👍

**Pull request merged**